nowplaying/input/observer.py

```diff
@@ -1,30 +1,28 @@
-#! /usr/bin/env python
-# -*- coding: utf-8 -*-
-
-__version__ = "$Revision$"
-# $Id$
-
 import logging
 import logging.handlers
 import os
 import time
+import warnings
 import xml.dom.minidom
+from abc import ABC, abstractmethod
 
 import isodate
 import pytz
 
-from nowplaying import show, track
+from nowplaying import show
 from nowplaying.show import client
+from nowplaying.track.track import DEFAULT_ARTIST, DEFAULT_TITLE, Track
 
 logger = logging.getLogger("now-playing")
 
-SHOW_NAME_KLANGBECKEN = "Klangbecken"
-SHOW_URL_KLANGBECKEN = "http://www.rabe.ch/sendungen/musik/klangbecken.html"
 
+class InputObserver(ABC):
```

Comment:
Just for reference, here are the [ABC docs](https://docs.python.org/3/library/abc.html).